p11-kit/rpc-server.c

```diff
@@ -175,18 +176,22 @@ proto_read_ulong_buffer (p11_rpc_message *msg,
 	/* Check that we're supposed to be reading this at this point */
 	assert (!msg->signature || p11_rpc_message_verify_part (msg, "fu"));
 
+	if (!p11_rpc_buffer_get_byte (msg->input, &msg->parsed, &buffer_is_null))
+		return PARSE_ERROR;
 	/* The number of ulongs there's room for on the other end */
 	if (!p11_rpc_buffer_get_uint32 (msg->input, &msg->parsed, &length))
 		return PARSE_ERROR;
 
 	*n_buffer = length;
 	*buffer = NULL;
 
-	/* If set to zero, then they just want the length */
-	if (length == 0)
+	/* If set to zero, then they just want the length for a *buffer = NULL */
```

Comment:
This comment looks misleading; how about:
```c
/* If buffer is NULL, the caller just want the length */
```

Reply:
thanks for the review, fixed.